pgjdbc/src/test/java/org/postgresql/test/ssl/SSLFactoryTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2024, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.ssl;
+
+import org.postgresql.PGProperty;
+import org.postgresql.core.SocketFactoryFactory;
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PSQLException;
+
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.InvocationTargetException;
+import java.security.NoSuchProviderException;
+import java.util.Properties;
+
+class SSLFactoryTest {
+
+  @Test
+  void TestDefault() throws Exception {
+    TestUtil.assumeSslTestsEnabled();
+
+    Properties props = new Properties();
+    PGProperty.SSL_FACTORY.set(props, "org.postgresql.ssl.DefaultJavaSSLFactory");
+
+    try {
+      SocketFactoryFactory.getSslSocketFactory(props);
+    } catch (Throwable t) {
+      /* ignore NoSuchProviderException */
+      if (!(t instanceof PSQLException)
+              || (!(t.getCause() instanceof InvocationTargetException))
+              || (!(t.getCause().getCause() instanceof PSQLException))
+              || (!(t.getCause().getCause().getCause() instanceof NoSuchProviderException))
+      ) {
```

Comment:
You are sure that cause will not be null at some point?

Reply:
Instanceof doesn't require a null check. The expression "null instancof PSQLException" returns false.